distributed/worker.py

```diff
@@ -267,6 +274,56 @@ def is_protected(self) -> bool:
         )
 
 
+class _UniqueTaskHeap:
```

Comment:
```suggestion
class UniqueTaskHeap(Collection[TaskState]):
```

Reply:
the class is referenced around the documentation and unit tested. I don't think it should start with _.

Reply:
technically, this would even classify as a `Set` which would bring a few mixin methods that could be useful in tests, i.e. the entire set logic would be applicable. thoughts?